appinventor/appengine/src/com/google/appinventor/client/explorer/commands/DeleteFileCommand.java

```diff
@@ -36,6 +70,158 @@ public boolean willCallExecuteNextCommand() {
 
   @Override
   public void execute(final ProjectNode node) {
+	  if (node instanceof YoungAndroidSourceNode) {
+  		new DeleteFormDialog((YoungAndroidSourceNode) node).center();
+  		
+	  } 
+	  else {
+  		  executionFailedOrCanceled();
+  		  throw new IllegalArgumentException("node must be a YoungAndroidProjectNode");
+	  }
+  }
+
+  /**
+   * Shows a confirmation dialog.
+   *
+   * @return {@code true} if the delete file command should be executed or
+   *         {@code false} if it should be canceled
+   */
+  protected boolean deleteConfirmation() {
+    return Window.confirm(MESSAGES.reallyDeleteFile());
+  }
+  
+  private class DeleteFormDialog extends DialogBox {
+    // UI elements
+    private final LabeledTextBox newNameTextBox;
+	private final CheckBox cb = new CheckBox(MESSAGES.checkBoxButton());
+    private final Set<String> otherFormNames;
+
+    DeleteFormDialog(final YoungAndroidSourceNode projectRootNode) {
+      super(false, true);
+      setStylePrimaryName("ode-DialogBox");
+      VerticalPanel contentPanel = new VerticalPanel();
+      setText(MESSAGES.removeFormButton());
+      final String prefix = "Screen";
+      final String abc;
+      final int prefixLength = prefix.length();
+      int highIndex = 0;
+      // Collect the existing form names so we can prevent duplicate form names.
+      otherFormNames = new HashSet<String>();
+      String defaultFormName = prefix + (highIndex + 1);
+      newNameTextBox = new LabeledTextBox(MESSAGES.formNameLabel());
+	    
+      String cancelText = MESSAGES.cancelButton();
+      String okText = MESSAGES.okButton();
+
+      // Keeps track of the total number of screens.
+      int formCount = otherFormNames.size() + 1;
+      if (formCount > MAX_FORM_COUNT) {
+        HorizontalPanel errorPanel = new HorizontalPanel();
+        HTML tooManyScreensLabel = new HTML(MESSAGES.formCountErrorLabel());
+        errorPanel.add(tooManyScreensLabel);
+        errorPanel.setSize("100%", "24px");
+        contentPanel.add(errorPanel);
+
+        okText = MESSAGES.addScreenButton();
+        cancelText = MESSAGES.cancelScreenButton();
+      }
+      Button cancelButton = new Button(cancelText);
+      cancelButton.addClickHandler(new ClickHandler() {
+        @Override
+        public void onClick(ClickEvent event) {
+          hide();
+          executionFailedOrCanceled();
+        }
+      });
+      
+      cb.setValue(true);
+      final Button okButton = new Button(okText);
+      okButton.addClickHandler(new ClickHandler() {
+        @Override
+        public void onClick(ClickEvent event) {
+    	   handleOkClick(projectRootNode);    
+        }
+      });
+   
+      if (projectRootNode instanceof YoungAndroidFormNode) {
```

Comment:
Whitespace here needs to be two spaces per indent.

Reply:
Added two whitespace per indent